More detailed explanation:

http://lists.pld-linux.org/mailman/pipermail/pld-devel-en/2012-December/023356.html

On Dec 1, 2012, at 10:55 AM, Jeffrey Johnson <n3npq@me.com> wrote:

> 
> I'm inclined atm to prefer the above actual behavior to "fudging" an
> extra +1 for "legacy compatible" behavior; I'm sure we disagree here.
> 
> Short answer: patch in an extra +1 (there will be two code paths in need
> of patching, check for symmetry as above) if you wish "legacy compatible" behavior.
> 

This is likely all that is needed (untested):

cvs diff rpm/lib/psm.c
Index: rpm/lib/psm.c
===================================================================
RCS file: /v/rpm/cvs/rpm/lib/psm.c,v
retrieving revision 2.399.2.5
diff -p -u -w -r2.399.2.5 rpm/lib/psm.c
--- rpm/lib/psm.c	19 Apr 2012 17:26:06 -0000	2.399.2.5
+++ rpm/lib/psm.c	1 Dec 2012 16:02:48 -0000
@@ -2755,7 +2755,7 @@ assert(psm->te != NULL);
 	    psm->scriptTag = RPMTAG_POSTIN;
 	    psm->progTag = RPMTAG_POSTINPROG;
 	    psm->sense = RPMSENSE_TRIGGERIN;
-	    psm->countCorrection = 0;
+	    psm->countCorrection = 1;
 
 	    if (!(rpmtsFlags(ts) & RPMTRANS_FLAG_NOPOST)) {
 		rc = (rpmRC) rpmpsmNext(psm, PSM_SCRIPT);

_______________________________________________
pld-devel-en mailing list
pld-devel-en@lists.pld-linux.org
http://lists.pld-linux.org/mailman/listinfo/pld-devel-en
